error.go

```diff
@@ -0,0 +1,353 @@
+package kafka
+
+import (
+	"fmt"
+	"io"
+)
+
+// Error represents the different error codes that may be returned by kafka.
+type Error int
+
+const (
```

Comment:
Yeah, I had NoError here initially, but then I was feeling like this was just stupid and useless, you just `!= 0` at the protocol level and `!= nil` in the program, this is so much more natural.